metadata-ingestion/src/datahub/ingestion/source/sql/bigquery.py

```diff
@@ -110,20 +147,38 @@ def __init__(self, config, ctx):
 
     def _compute_big_query_lineage(self) -> None:
         if self.config.include_table_lineage:
-            try:
-                _clients: List[GCPLoggingClient] = self._make_bigquery_client()
-                log_entries: Iterable[AuditLogEntry] = self._get_bigquery_log_entries(
-                    _clients
-                )
-                parsed_entries: Iterable[QueryEvent] = self._parse_bigquery_log_entries(
-                    log_entries
-                )
-                self.lineage_metadata = self._create_lineage_map(parsed_entries)
-            except Exception as e:
-                logger.error(
-                    "Error computing lineage information using GCP logs.",
-                    e,
-                )
+            if self.config.use_exported_bigquery_audit_metadata:
+                self._compute_bigquery_lineage_via_exported_bigquery_audit_metadata()
+            else:
+                self._compute_bigquery_lineage_via_gcp_logging()
+
+    def _compute_bigquery_lineage_via_gcp_logging(self) -> None:
+        try:
+            _clients: List[GCPLoggingClient] = self._make_bigquery_client()
+            log_entries: Iterable[AuditLogEntry] = self._get_bigquery_log_entries(
+                _clients
+            )
+            parsed_entries: Iterable[QueryEvent] = self._parse_bigquery_log_entries(
+                log_entries
+            )
+            self.lineage_metadata = self._create_lineage_map(parsed_entries)
+        except Exception as e:
+            logger.error(
+                "Error computing lineage information using GCP logs.",
+                e,
+            )
+
+    def _compute_bigquery_lineage_via_exported_bigquery_audit_metadata(self) -> None:
+        _client: BigQueryClient = BigQueryClient(project=self.config.project_id)
+        exported_bigquery_audit_metadata: Iterable[
+            BigQueryAuditMetadata
+        ] = self._get_exported_bigquery_audit_metadata(_client)
+        parsed_entries: Iterable[
+            QueryEvent
+        ] = self._parse_exported_bigquery_audit_metadata(
+            exported_bigquery_audit_metadata
+        )
+        self.lineage_metadata = self._create_lineage_map(parsed_entries)
```

Comment:
Do we not need to put this in its own try-catch block? There could be failures like service-account not having access to the sink table or other processing failures.

Reply:
Acknowledged this comment. Please take a look! 🙏